src/lib/core/Optional.h

```diff
@@ -241,13 +242,48 @@ class Optional
     }
 
 private:
-    bool mHasValue;
-    union Value
+    // A container of bool + value (without constructor/destructor) when the underlying
+    // type has a trivial destructor
+    class TrivialDestructor
+    {
+    public:
+        bool mHasValue = false;
+        union Value
+        {
+            Value() {}
+            char _unused;
```

Comment:
Please document why that char is there.  Because seems like it just makes things larger when T is a one-byte thing or something....

Reply:
Remnant from when I was investigating how to make this code bend to my will (given that I cannot SFNIAE destructors). Removed the unused bits. Size-wise I believe chars cannot make things larger because values in C++ need unique addresses so they have no 0-sized structures (and this is a union, so it should have at least 1 byte of storage...). In any case, gone and tests still pass.